src/hypercube.rs

```diff
@@ -0,0 +1,66 @@
+//! This module defines basic types related to Boolean hypercubes.
+#![allow(unused)]
+use std::marker::PhantomData;
+
+use crate::utils::*;
+/// There's some overlap with polynomial.rs.
+use ff::PrimeField;
+use itertools::Itertools;
+
+#[derive(Debug)]
+pub(crate) struct BooleanHypercube<F: PrimeField> {
```

Comment:
This type might be unnecessary since multilinear polynomials are going to be stored in evaluation form over the Boolean hypercube. In particular, given a vector z that represents evaluations of multilinear polynomial \widetilde{z}, if we need the evaluation of \widetilde{z}(x) where x is an appropriate bit string represented with usize, we could simply return z[x]. In essence, we are treating bitstrings as indexes into the vector (we can see examples of this in src/spartan/polynomial.rs). This would avoid two separate representations for polynomials.